payload/builder/payload.go

```diff
@@ -204,9 +216,6 @@ func (pb *PayloadBuilder) getPayload(
 	if err != nil {
 		return nil, err
 	}
-	if envelope == nil {
-		return nil, ErrNilPayloadEnvelope
-	}
```

Comment:
nit: this error was never checked

Reply:
- nilaway still passes as is, the error is never really used
- ive consolidated nil payloads with a unilateral error that is also shared by the EL. see [here](https://github.com/berachain/beacon-kit/pull/2949#discussion_r2433189050) for details